### DISCUSSION TITLE
apis error data is empty

### DISCUSSION THREAD

**Discussion by Oggy4th** [pocketbase/pocketbase#2253] [Q&A] [answered]
`return apis.NewNotFoundError("didn't find", err)`

it's returning
```
{
  "code": 404,
  "message": "Didn't find.",
  "data": {}
}
```

Expecting `sql: no rows in result set` inside `data`

why is it empty?

**Comment by ganigeorgiev** [comment, marked as answer]
Only errors that satisfy the `validation.Errors` interface will be returned in the `data` prop, because otherwise the error could contain sensitive information and could be a security issue.

If you want to set any value you can modify the `Data` field map manually, but I don't recommend it especially for raw sql errors:

```js
apiErr := apis.NewNotFoundError("didn't find", err)

apiErr.Data["something"] = err

return apiErr
```

Alternatively, if you want to show the raw error message, you can do something like:

```go
return apis.NewNotFoundError(err.Error(), nil)
```

But again, return raw SQL errors only for admin level endpoints or when you really need it.
Note that even if you return a raw SQL error to regular app users, they will not be able to do anything about it and it is better to show a human friendly message instead.
